### ISSUE TITLE
Incorrect SemVer style checking.

### ISSUE THREAD

**Issue by AndreiPashkin** [mtkennerly/dunamai#12]
Let's take an example version string:
`'0.0.2-post.41+1.5ccabe4.dirty`

And let's check it against the [current SemVer pattern][1] from the project's codebase: https://regex101.com/r/1i2CTA/1, it doesn't match.
But if take the regex from semver.org: https://regex101.com/r/C2Hc9l/1, it would match.

It seems like SemVer regex should be updated in Dunamai.

[1]: https://github.com/mtkennerly/dunamai/blob/e473d6816046805e2ecb8aefc91c4a58324b1a9b/dunamai/__init__.py#L36

**Comment by mtkennerly** [comment]
Thanks for reporting this! It looks like Dunamai has an incorrect qualifier here (`?` instead of `*`):

```diff
diff --git a/dunamai/__init__.py b/dunamai/__init__.py
index 398c12e..5510ddc 100644
--- a/dunamai/__init__.py
+++ b/dunamai/__init__.py
@@ -33,7 +33,7 @@ _VERSION_PATTERN = r"""
 # PEP 440: [N!]N(.N)*[{a|b|rc}N][.postN][.devN][+<local version label>]
 _VALID_PEP440 = r"^(\d!)?\d+(\.\d+)*((a|b|rc)\d+)?(\.post\d+)?(\.dev\d+)?(\+.+)?$"
 _VALID_SEMVER = (
-    r"^\d+\.\d+\.\d+(\-[a-zA-z0-9\-]+(\.[a-zA-z0-9\-]+)*)?(\+[a-zA-z0-9\-]+(\.[a-zA-z0-9\-]+)?)?$"
+    r"^\d+\.\d+\.\d+(\-[a-zA-z0-9\-]+(\.[a-zA-z0-9\-]+)*)?(\+[a-zA-z0-9\-]+(\.[a-zA-z0-9\-]+)*)?$"
 )
 _VALID_PVP = r"^\d+(\.\d+)*(-[a-zA-Z0-9]+)*$"
```

I'll release a fix for this soon.

**Comment by mtkennerly** [comment]
This is now fixed in v1.5.3.
